### PR TITLE
Alpha stage api fixes

### PR DESCRIPTION
**Pull request by duskfallcrew**
Reason for Merging wihtotu Testing:

Runpod doesn't like to restart easily, and i have more credits on runpod... *SIGHS*

FIXES AFAIK:

SD-Scripts:
- Version Support upgrade
- API Layer Python - Updated
- API Layer React/TS - Updated
- UI/UX For Upgrades - Updated 

Lycoris:

- Version Bump
- Python Api layer
- TS/React API Layer
- UI/UX For Ugprades

Security Layer via Snyk

- API Layer Fixes
- Vulnerability Fixes
- Path Validation

What MIGHT BE BROKEN and will get fixed within 5 hours or less of this PR Merge:

- package lock
- A few TS files claude didn't finish in the security run cause i ran out of time :D 

HEY ANTHROPIC: :3 If you're listening? - Wanna semi sponsor this as a "Case of Open Source on the low end, that's providing a service"

By sponsor i just mean:
Give discount to the current sub tier i'm on - I don't ask for max LOL, i'm not running a huge 200k star repo. 

## Summary by CodeRabbit

* **New Features**
  * Added support for Anima, HunyuanImage-2.1 and Chroma workflows; new minimal inference and training scripts; ABBA LoRA option and Save actions in training UI.

* **Improvements**
  * FP8 weight optimizations, stronger path validation to prevent path traversal, improved offload/memory handling, expanded GPU compatibility guidance, and UI tweaks (dataset dropdown label, tag commit behavior, settings handling).

* **Documentation**
  * Major README and training/inference guides expanded with model-specific docs and installation/requirements updates.

### CODE REVIEW

**Comment by duskfallcrew** [comment]
Claude Code Review probably needs to get shut off because it's running on my normal sub rather than extra, so i need to be aware of that -- it'll fail every single time because if i'm not with any token space left it'll just fail XD

**Comment by duskfallcrew** [comment]
@coderabbitai help

File: frontend/app/api/config/presets/[id]/route.ts [hunk 1]
```diff
@@ -10,10 +10,10 @@ import path from 'path';
 
 export async function GET(
   request: NextRequest,
-  { params }: { params: { id: string } }
+  { params }: { params: Promise<{ id: string }> }
 ) {
   try {
-    const { id } = params;
+    const { id } = await params;
```
[the remark this answers is not in the document]
Comment (duskfallcrew) [reply frontend/app/api/config/presets/[id]/route.ts]: You're making me want to go get lunch and then acutally wait for Claude instead of ADHD'ing into testing this before 2pm... I realize i could do it myself but .. i'm gonna be dead honest xD stuff like this makes my brain hurt. So cool, i'll get this one checked with claude in a while

File: trainer/derrian_backend/sd_scripts/library/custom_offloading_utils.py [hunk 2]
```diff
@@ -241,10 +261,85 @@ def wait_for_block(self, block_idx: int):
...
+    Returns:
+        A wrapped function that offloads inputs to CPU and moves outputs back to the specified device.
+    """
+
+    def wrapper(orig_func: Callable) -> Callable:
+        def custom_forward(*inputs):
+            nonlocal device, orig_func
+            cuda_inputs = to_device(inputs, device)
+            outputs = orig_func(*cuda_inputs)
+            return to_cpu(outputs)
```
[the remark this answers is not in the document]
Comment (duskfallcrew) [reply trainer/derrian_backend/sd_scripts/library/custom_offloading_utils.py]: Weird, agian vendored code BUT i'l double check with claude

File: trainer/derrian_backend/sd_scripts/finetune/tag_images_by_wd14_tagger.py [hunk 3]
```diff
@@ -191,35 +234,95 @@ def main(args):
...
-            if source in general_tags:
-                general_tags[general_tags.index(source)] = target
-            elif source in character_tags:
-                character_tags[character_tags.index(source)] = target
-            elif source in rating_tags:
-                rating_tags[rating_tags.index(source)] = target
+            if source in tags:
+                tags[tags.index(source)] = target
+
+        return tags
```
[the remark this answers is not in the document]
Comment (duskfallcrew) [reply trainer/derrian_backend/sd_scripts/finetune/tag_images_by_wd14_tagger.py]: Please do! that would be wonderful!

File: trainer/derrian_backend/lycoris/lycoris/kohya.py [hunk 4]
```diff
@@ -688,41 +881,191 @@ def apply_max_norm_regularization(self, max_norm_value, device):
...
+                continue
 
+            lr_description = f"{comp_type}{comp_idx if comp_type == 'textencoder' else ''}{' Ortho' if is_ortho_group else ''}{' plus' if is_lora_plus else ''}"
+
+            group_dict = {
+                'params': params,
+                'lr_description': lr_description,
+                'is_ortho_group': is_ortho_group,
+                'is_lora_plus_group': is_lora_plus,
+                'lr': torch.tensor(current_lr),
```
[the remark this answers is not in the document]
Comment (duskfallcrew) [reply trainer/derrian_backend/lycoris/lycoris/kohya.py]: Right but it seems to be a thing we could still poke, because even though we're vendoring, if it breaks on our side i feel like we could still fix it - and then apply the fix in our fork on the other repo and share the fix otherwise - but it COULD be that the main repo because our fork is a customized fork that our setup was using -- that Kohaku's main code might be fine so i'l lcheck with cluade on that too

File: diagnose.bat [hunk 5]
```diff
@@ -47,6 +47,14 @@ pause
 exit /b 1
 
 :RUN_DIAGNOSTIC
+
+REM Activate virtual environment if it exists (so packages are visible)
+if exist ".venv\Scripts\activate.bat" (
+    echo Activating virtual environment...
+    call .venv\Scripts\activate.bat
+    set PYTHON_CMD=python
+)
```
[the remark this answers is not in the document]
Comment (duskfallcrew) [reply diagnose.bat]: Yea it's something i'll fix agai nlater on part of it's because we have to have multiple layered venv's cause of editable scripts in the vendored code The key point might also be that users shouldnt' ahve to diagnose upon succesful run XD
